src/plugins/cordova-plugin-media/app-host-handlers.js

```diff
@@ -2,281 +2,106 @@
-        }
-    } else {
-         console.error && console.error("Received Media.onStatus callback for unknown media :: " + id);
-    }
-};
-
-document.addEventListener("deviceready", function () {
-    window.Media = Media;
-}, false);
-
 module.exports = function (messages) {
 
-    function emptyHandler() {}
+    var Player = require('./player');
 
     return {
         'Media': {
-            // Despite we fully override Media object on client side
-            // there is still one exec call below which is invoked during
-            // platform proxy initialization on android and windowsphone
-            'messageChannel': emptyHandler
+            'messageChannel': function (success, fail, args) {
+                // ignore, this is only available for "android", "amazon-fireos" and "windowsphone"
+                // platforms, any other access to onStatus callback is done through
+                // window.Media.onStatus
+            },
+            'create': function (success, fail, args) {
+                // args[0]: id, args[1]: src
+                Player.createIfNeeded(args[0], args[1]);
+            },
+            'startPlayingAudio': function (success, fail, args) {
+                var player = Player.createIfNeeded(args[0], args[1]);
+                player.startPlaying(args[1]);
+            },
+            'stopPlayingAudio': function (success, fail, args) {
+                var player = Player.getById(args[0]);
+                if (player) {
+                    player.stopPlaying();
+                }
+            },
+            'seekToAudio': function (success, fail, args) {
+                var player = Player.getById(args[0]);
+                if (player) {
+                    player.seekToAudio(args[1]);
+                }
+            },
+            'pausePlayingAudio': function (success, fail, args) {
+                var player = Player.getById(args[0]);
+                if (player) {
+                    player.pausePlaying();
+                }
+            },
+            'getCurrentPositionAudio': function (success, fail, args) {
+                var player = Player.getById(args[0]);
+
+                if (player) {
+                    player.getCurrentPosition(success, fail);
+                } else {
+                    success(-1);
+                }
+            },
+            'startRecordingAudio': function (success, fail, args) {
+                var player = Player.createIfNeeded(args[0]);
+                player.startRecord(args[1]);
+            },
+            'stopRecordingAudio': function (success, fail, args) {
+                var player = Player.getById(args[0]);
+                if (player) {
+                    player.stopRecord(true);
+                }
+            },
+            'pauseRecordingAudio': function (success, fail, args) {
+                var player = Player.getById(args[0]);
+                if (player) {
+                    player.stopRecord(false);
```

Comment:
According to the implementation [here](https://github.com/apache/cordova-plugin-media/blob/master/src/android/AudioHandler.java#L120), this should be `true` for `stopRecordingAudio` and `false` for `pauseRecordingAudio`. This is just to follow the other implementation, but when we support recording, we can change it :)